lib/stream_data.ex

```diff
@@ -65,15 +75,26 @@ defmodule StreamData do
   defmodule TooManyDuplicatesError do
     defexception [:message]
 
+    # FIX: A good error message says what went wrong, why it went wrong
+    # and possibly what can be done to fix it.
     def exception(options) do
       %__MODULE__{message: "too many duplicates: #{inspect(options)}"}
     end
   end
 
+  # TODO: All of the docs tells what happens on shrinking
+  # *before* it shows examples. I would reverse that. I am
+  # not interested on shrinking before I learn what the function
+  # does and how it works.
+
   ### Minimal interface
 
   ## Helpers
 
+  # QUESTION: Should we underscore those? We typically avoid
+  # such functions in STDLIB modules because import StreamData
+  # would bring them in. Alternatively we can make them public.
```

Comment:
Mh great point. I would start with underscoring maybe because I don't want users to know how a generator works internally for now. Wdyt?